pkg/local_object_storage/engine/delete.go

```diff
@@ -47,24 +47,28 @@ func (e *StorageEngine) delete(prm *DeletePrm) (*DeleteRes, error) {
 		defer elapsed(e.metrics.AddDeleteDuration)()
 	}
 
-	shPrm := new(shard.InhumePrm)
-	existsPrm := new(shard.ExistsPrm)
+	var shPrm shard.InhumePrm
```

Comment:
what for?

Reply:
Because after removal of return value from `With*` functions, it can be allocated on stack.

Reply:
And it will stay correct if we decide to parallelize iteration over shards.

Reply:
> Because after removal of return value from With* functions, it can be allocated on stack.

Isn't it still true for a one var declaration before the callback call?

> And it will remain correct if we decide to parallelize iteration over shards.

agree here, moved inside the callback